### PR TITLE
Login Page Enhancement

### PR DESCRIPTION
**Pull request by urvashi-code1255**
## Issue that this pull request solves

 Closes: #180

## Proposed changes
 * Make the Sign In button similar to that in contact-us page having the same hovering effect and look.
 * After click on input box, currently we see a black border, replace that with the bolder box shadow
 * Make the hide/show icon feature working
 * Make the cursor pointer to the hide/show icon

### Brief description of what is fixed or changed
* I have made the sign up button as same as the submit button on contact us page.
* Removed the black border of the input boxes and applied a bolder box shadow.
* I have made the hide/show icon feature working and its cursor pointer.

## Types of changes
* Javascript
* React.js
* CSS

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@urvashi-code1255 the hide/show password feature isn't working properly. Also Please make the witdth of `Sign In` button smaller, maybe same as that in the `contact-us page`

**Comment by urvashi-code1255** [comment]
@Kajol-Kumari So, Sign in button should be smaller and in middle?

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari So, Sign in button should be smaller and in middle?

yes! and just now  noticed that the changes in this page is effecting the contact-us page as well(The input box shadow after click), please fix that

**Comment by urvashi-code1255** [comment]
> > @Kajol-Kumari So, Sign in button should be smaller and in middle?
> 
> yes! and just now noticed that the changes in this page is effecting the contact-us page as well(The input box shadow after click), please fix that

@Kajol-Kumari Do you mean that before my changes there was black border on clicking the input field but now its blue?

**Comment by urvashi-code1255** [comment]
@Kajol-Kumari I did the requested changes.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @urvashi-code1255 the toggle feature is ok but needs enhacements
1. It should be replaced with an open eye when we would be able to see the password and vice-versa for hidden.
2. The box shadow after click doesn't look that good, maybe try black, instead of grey or any other color that might look good.
3. Also keep your branch updated in order to avoid any kind of conflicts

**Comment by urvashi-code1255** [comment]
Sure @Kajol-Kumari

**Comment by urvashi-code1255** [comment]
@Kajol-Kumari I am using this code for the change in icons 
Capture1
Capture2
But i m getting this error, could you please tell me what i m doing wrong? Or can you suggest some other way to do this?
Capture3

**Comment by urvashi-code1255** [comment]
Capture
Do, this border color looks Okay @Kajol-Kumari  ?

**Comment by Surajbokde** [comment]
> Capture
> Do, this border color looks Okay @Kajol-Kumari ?

@urvashi-code1255 try with  gradient (pink to blue)

**Comment by urvashi-code1255** [comment]
> @Kajol-Kumari I am using this code for the change in icons 
> Capture1
> Capture2
> But i m getting this error, could you please tell me what i m doing wrong? Or can you suggest some other way to do this?
> Capture3
> 

@Kajol-Kumari please tell me about this also.

**Comment by Kajol-Kumari** [comment]
Hey @urvashi-code1255 you need to push the changes as I am not able to see any suggested changes in here

**Comment by urvashi-code1255** [comment]
> Hey @urvashi-code1255 you need to push the changes as I am not able to see any suggested changes in here

@Kajol-Kumari I have pushed the changes please check ..

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @urvashi-code1255 password toggle feature is still not working as expected, please fix it! Also Please remove the commented code if it's not required anymore

**Comment by urvashi-code1255** [comment]
@Kajol-Kumari actually I have written the code and it should work fine. But it's not working. Could you please give your review for what I m doing wrong

File: frontend/src/components/auth/login/login.js [hunk 1]
```diff
@@ -175,4 +192,30 @@ const Login = () => {
   );
 };
 
+// const togglePassword = document.querySelector('#togglePassword');
```
Comment (Kajol-Kumari) [inline frontend/src/components/auth/login/login.js:195]: Remove this whole commente dsection from,[here](https://github.com/HITK-TECH-Community/Community-Website/pull/188/files#diff-d5c81cb0d6ea549df508bdd0e919acef604720ffcee342acea1c1170ed52f506R195) to [here](https://github.com/HITK-TECH-Community/Community-Website/pull/188/files#diff-d5c81cb0d6ea549df508bdd0e919acef604720ffcee342acea1c1170ed52f506R220)

File: frontend/src/components/auth/login/login.js [hunk 2]
```diff
@@ -1,8 +1,22 @@
 import React, { useReducer, useEffect } from "react";
 import "./login.css";
 
-//state type
+const togglePassword = document.querySelector('#togglePassword');
+const password = document.querySelector('#password');
 
+// window.onload=function () {
```
Comment (Kajol-Kumari) [inline frontend/src/components/auth/login/login.js:7]: uncomment this line

File: frontend/src/components/auth/login/login.js [hunk 3]
```diff
@@ -1,8 +1,22 @@
...
+if (togglePassword) {
+  togglePassword.addEventListener('click', function (e) {
+    // toggle the type attribute
+    // if(password){
```
Comment (Kajol-Kumari) [inline frontend/src/components/auth/login/login.js:11]: uncomment this line

File: frontend/src/components/auth/login/login.js [hunk 4]
```diff
@@ -1,8 +1,22 @@
...
+    const type = password.getAttribute('type') == 'password' ? 'text' : 'password';
+    password.setAttribute('type', type);
+    // toggle the eye slash icon
+    this.classList.toggle('fa-eye-slash');
+    // }
```
Comment (Kajol-Kumari) [inline frontend/src/components/auth/login/login.js:16]: uncomment this

File: frontend/src/components/auth/login/login.js [hunk 5]
```diff
@@ -1,8 +1,22 @@
...
+  });
+};
```
Comment (Kajol-Kumari) [inline frontend/src/components/auth/login/login.js:18]: add a closing brace(`}`) before this line

File: frontend/src/components/contact/contact_us.css [hunk 6]
```diff
@@ -89,11 +89,11 @@
   color: #777777;
 }
 
-.contact-input input:focus,
+/* .contact-input input:focus,
```
Comment (Kajol-Kumari) [inline frontend/src/components/contact/contact_us.css:92]: undo unnecessary change in this file

File: frontend/src/components/auth/login/login.css [hunk 7]
```diff
@@ -66,6 +65,14 @@
   box-shadow: inset 2px 2px 5px #888888, inset -2px -2px 5px #ffffff;
 }
 
+.inputLogin:focus{
+  box-shadow: inset 3px 2px 5px #050142, inset -3px -2px 5px #e005d5;
```
Comment (Kajol-Kumari) [inline frontend/src/components/auth/login/login.css:69]: change this to `box-shadow: inset 2px 2px 5px #1E488F, inset -2px -2px 5px #1E488F;`

File: frontend/src/components/auth/login/login.css [hunk 8]
```diff
@@ -77,10 +84,14 @@
 }
 
 .login-input input:focus {
-  border-color: #1863ff !important ;
+  /* border-color: #1863ff !important ; */
```
Comment (Kajol-Kumari) [inline frontend/src/components/auth/login/login.css:87]: remove this commented line

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @urvashi-code1255 Along with styling your code has a lot of issues. please fix them all and the code would work as expected :)

Please use proper formatting and commenting and try to write clean code!
Happy learning :)

File: frontend/src/components/auth/login/login.css (hunk 8 above)
Comment (urvashi-code1255) [reply frontend/src/components/auth/login/login.css]: @Kajol-Kumari I did all the changes suggested by you. Also, before commenting all that code, I also did the same thing which you suggested but it was not working. Also, After doing changes you said, now also its not working from my end. Please check.

Thank you

File: frontend/src/components/auth/login/login.js [hunk 9]
```diff
@@ -1,8 +1,25 @@
 import React, { useReducer, useEffect } from "react";
 import "./login.css";
 
-//state type
```
Comment (Kajol-Kumari) [inline frontend/src/components/auth/login/login.js:4]: @urvashi-code1255 replace the lines from [here](https://github.com/HITK-TECH-Community/Community-Website/pull/188/files#diff-d5c81cb0d6ea549df508bdd0e919acef604720ffcee342acea1c1170ed52f506L4) to [here](https://github.com/HITK-TECH-Community/Community-Website/pull/188/files#diff-d5c81cb0d6ea549df508bdd0e919acef604720ffcee342acea1c1170ed52f506R20) with 

```
// Show/Hide Functionality.
window.onload = function () {
  const togglePassword = document.querySelector('#togglePassword');
  const password = document.querySelector('#password');
  if (togglePassword) {
    togglePassword.addEventListener('click', function (e) {
      // toggle the type attribute
      if (password) {
        const type = password.getAttribute('type') == 'password' ? 'text' : 'password';
        password.setAttribute('type', type);
        // toggle the eye slash icon
        this.classList.toggle('fa-eye-slash');
      }
    });
  }
};
```

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@urvashi-code1255 I recommend you to google the errors and try fixing them. These are minor DOM related issues.I have attached the solution for this PR but try to search it. It will enhance your searching skill as well :)

File: frontend/src/components/auth/login/login.js (hunk 9 above)
Comment (urvashi-code1255) [reply frontend/src/components/auth/login/login.js:4]: Oh I have to make the variables also after loading the window. Thank you so much @Kajol-Kumari. Actually I have searched a lot but this thing hasn't come to my mind. Thank you :)

**Comment by urvashi-code1255** [comment]
@Kajol-Kumari I did the changes . Now, its working from my end. Please check...

**Review by Kajol-Kumari — APPROVED** [review]
Thanks for incorporating the suggested changes @urvashi-code1255 
LGTM :+1:

**Comment by Kajol-Kumari** [comment]
@urvashi-code1255 please make sure that all your merged PRs have appropriate labels added to it. Incase we miss any, let us know.
Happy Learing :)

**Comment by urvashi-code1255** [comment]
@Kajol-Kumari I am participating in Swoc and dwoc

**Comment by urvashi-code1255** [comment]
@Kajol-Kumari @jackfrost13 I am swoc and dwoc participant.
